src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -85,4 +131,338 @@ public void requestMessageFromIris(IrisSession session) {
+         * @param problemStatement The problem statement to be truncated
+         * @return Problem statement without specific markdown marks
+         */
+        private static String removeHrs(String problemStatement) {
+            String temp = problemStatement;
+            temp = temp.replace("---", "");
+            temp = temp.replace("***", "");
+            temp = temp.replace("- - - -", "");
+            temp = temp.replace("~~~", "");
+            return temp;
+        }
+
+        /**
+         * Removes all uml notation (@startuml * @enduml) from the problem statement
+         *
+         * @param problemStatement The problem statement to be truncated
+         * @return Problem statement without uml notation
+         */
+        private static String removeUML(String problemStatement) {
+            String temp = problemStatement;
+            String pString = "(@startuml)(.*?)(@enduml)";
+            Pattern pattern = Pattern.compile(pString, Pattern.MULTILINE);
+            Matcher matcher = pattern.matcher(problemStatement);
+            while (matcher.find()) {
+                temp = temp.replace(matcher.group(0), "");
+            }
+            return temp;
+        }
+
+        /**
+         * Removes all tasks ([task][*](*)) from the problem statement but keeps the task name
+         *
+         * @param problemStatement The problem statement to be truncated
+         * @return Problem statement without tasks
+         */
+        private static String removeTasks(String problemStatement) {
+            String temp = problemStatement;
+            String pString = "(\\[task]\\[)(.*?)(]\\()(.*?)(\\))";
+            Pattern pattern = Pattern.compile(pString, Pattern.MULTILINE);
+            Matcher matcher = pattern.matcher(problemStatement);
+            while (matcher.find()) {
+                temp = temp.replace(matcher.group(0), matcher.group(2));
+            }
+            return temp;
+        }
+
+        /**
+         * Removes all opening divs (<div *>) from the problem statement
+         *
+         * @param problemStatement The problem statement to be truncated
+         * @return Problem statement without opening div tags
+         */
+        private static String removeDiv(String problemStatement) {
+            String temp = problemStatement;
+            String pString = "(<div)(.*?)(>)";
+            Pattern pattern = Pattern.compile(pString, Pattern.MULTILINE);
+            Matcher matcher = pattern.matcher(problemStatement);
+            while (matcher.find()) {
+                temp = temp.replace(matcher.group(0), " ");
+            }
+            return temp;
+        }
+
+        /**
+         * Removes all opening spans (<span *>) from the problem statement
+         *
+         * @param problemStatement The problem statement to be truncated
+         * @return Problem statement without opening span tags
+         */
+        private static String removeSpan(String problemStatement) {
```

Comment:
I saw some duplicate code in the methods (e.g. removeStyle, removeDiv, etc.). Most of them just differ in the pString. I think you could combine them with a method that takes in the pString (and group index) as additional parameters and does the rest.